core/src/main/java/com/datastax/oss/driver/internal/core/type/codec/StringCodec.java

```diff
@@ -74,7 +95,11 @@ public String decode(@Nullable ByteBuffer bytes, @NonNull ProtocolVersion protoc
     } else if (bytes.remaining() == 0) {
       return "";
     } else {
-      return new String(Bytes.getArray(bytes), charset);
+      try {
+        return decoder.decode(bytes.duplicate()).toString();
```

Comment:
Not a fan of the .duplicate() here, but without it some other tests start failing. Maybe a maintainer could make the necessary additional changes in case the duplicate() is indeed unwanted

Reply:
It was definitely wrong, so I went with the FastThreadLocal like in the TimestampCodec

Reply:
No it's the right way to do it. The codec operates directly on the bytebuffer stored in the data structure (`Row`, `UdtValue`, etc.), and the contract is that getters don't consume the data, users should be able to call `getString`, `getBytesUnsafe`, etc, multiple times on the same column.
Other codec implementations like `ListCodec` also duplicate the buffer.